### ISSUE TITLE
[Foward Port] UI doesn't update RKE1 provisioned cluster status after making changes to it

### ISSUE THREAD

**Issue by richard-cox** [rancher/dashboard#8250]
2.6.11 issue: rancher/dashboard#8237
Forward port rancher/dashboard#8249

**Comment by anupama2501** [comment]
Re opening as I still notice the issue on v2.7-head `2b2ed44` and dashboard commit `release-2.7.2 0dcdb177b`

- As an admin created 2 standard users - u1 and u2
- As a cluster owner, created an rke1 node driver cluster with 1 etcd+cp pool and 1 worker pool on k8s 1.23
- Add u2 as cluster member in all the downstream cluster.
- Edit the config and changed the value of ACE to disabled and the status is not updated until we refresh the UI.
- Once the status is updated on a UI refresh from active to upgrading to active for cluster owner, for cluster member the status is stuck on `updating`
Expected the status to update from active to update to active without needing the UI refresh.

<img width="906" alt="2023-03-24_13-42-50" src="https://user-images.githubusercontent.com/38144301/227637484-dc3552ea-b7e0-4900-9678-64aba8d47bbe.png">

**Comment by richard-cox** [comment]
I've run through the following a few times (which is similar but not exactly the same as the flow described in https://github.com/rancher/dashboard/issues/8250#issuecomment-1483392130) and given the scenario the cluster member isn't getting the change in mgmt cluster `metadata.state` that admin and cluster owners are.

In my tests i have three browsers
- browser 1 - brave - admin - publicly accessible rancher
- browser 2 - brave - user1 (cluster owner) - local dashboard build
- browser 3 - chrome - user2 (cluster member) - local dashboard build

1. b1 / admin creates and rke1 1.23 DO cluster
1. b1 / admin add user1 as cluster owner
1. b1 / admin add user2 as cluster member
1. confirm all three browsers / users cluster management page's show the cluster as `active`
1. b1 / admin edits cluster `Authorized Cluster Endpoint` (toggle between Enabled / Disabled)
1. Monitor `/v1` socket messages for `management.cattle.io.cluster` representation of cluster
    - b1 / admin - receives `resoure.change` messages for mgmt cluster showing required changes in `metadata.state`
    - b2 / user1 - same as b1 / admin
    - b3 / user2 - doesn't receive any `resource.change` for mgmt cluster

I've tested this flow twice, both with the same result. I did spot the cluster status in both b1 and b2 was slow to update in some cases, but would eventually become right.

Interesting points
- Is there something special about changing the ACE setting that would mean cluster members don't get the update?
- The cluster member gets updates to the `counts` resource, but no `resource.start` / `resource.remove` messages. I haven't investigated this, it could be the `counts` message payload hasn't changed (aka resource.change is sent even though they can't see changed resource).

Issue Progression
@anupama2501 I think the issue in https://github.com/rancher/dashboard/issues/8250#issuecomment-1483392130 is a new, separate issue than the one described in this one (aka https://github.com/rancher/dashboard/issues/8237). Could you create a gh issue for your comment over at `rancher/rancher` and i'll then add info from this comment?

**Comment by anupama2501** [comment]
@richard-cox I ran a few tests today and noticed that this is not the case only when w update the ACE endpoints but make any changes to the cluster.yaml. 
For example, I ran the following tests and noticed that for each of these tests only cluster owners and admins get updated status but not cluster members:
If we upgrade the k8s version from 1.24 to 1.25 
If we disable the `recurring snapshots`
If we enable/disable the ACE endpoints.

Since this seems specific to cluster members only, I'll log a ticket for that.

Verified following scenarios and the status is updated for cluster owner and admin at the sane time without needing to refresh the UI.

Rancher version: v2.7-head `3a42367 `
Dashboard version `release-2.7.2 4689b621e`

1. As an admin, create an rke1 node driver cluster on k8s v1.24
2. Created 1 standard users and add them as cluster owner.
3. As a cluster owner, scale up nodes - status changes at the same time for all three users cluster member, cluster owner and admin
4. Upgrade k8s version from 1.24 to 1.25 and the status is updated for both admin and CO
5. Scale down nodes - the status is updated for both admin and CO and CM as well at the same time.
6. Take etcd snapshots and the status is updated .
